### PR TITLE
update native contract adapt to neovm 

### PR DESCRIPTION
**Pull request by tanZiWen**

### CODE REVIEW

File: vm/neovm/func_array.go [hunk 1]
```diff
@@ -151,8 +151,13 @@ func opAppend(e *ExecutionEngine) (VMState, error) {
 	if value, ok := newItem.(*types.Struct); ok {
 		newItem = value.Clone()
 	}
-	itemArr := PopArray(e)
-	itemArr = append(itemArr, newItem)
+	items := PopStackItem(e)
+	if item, ok := items.(*types.Array); ok {
+		item.Add(item)
```
Comment (laizy) [inline vm/neovm/func_array.go]: seems interesting. add item to it self?

**Review by laizy — CHANGES_REQUESTED** [review]

File: core/store/ledgerstore/ledger_store.go [hunk 2]
```diff
@@ -786,31 +785,20 @@ func (this *LedgerStoreImp) PreExecuteContract(tx *types.Transaction) (*sstate.P
 		Config:     config,
 		Store:      this,
 		CloneCache: storage.NewCloneCache(this.stateStore.NewStateBatch()),
-		Code:       invoke.Code,
 		Gas:        math.MaxUint64,
 	}
 
 	//start the smart contract executive function
-	result, err := sc.Execute()
+	engine, _ := sc.NewExecuteEngine(invoke.Code)
```
Comment (Arbio5zt) [inline core/store/ledgerstore/ledger_store.go]: miss error catch.

File: core/store/ledgerstore/tx_handler.go [hunk 3]
```diff
@@ -97,10 +88,12 @@ func (self *StateStore) HandleInvokeTransaction(store store.LedgerStore, stateBa
 	tx *types.Transaction, block *types.Block, eventStore scommon.EventStore) error {
 	invoke := tx.Payload.(*payload.InvokeCode)
 	txHash := tx.Hash()
-	code := invoke.Code.Code
+	code := invoke.Code
 	sysTransFlag := bytes.Compare(code, ninit.COMMIT_DPOS_BYTES) == 0 || block.Header.Height == 0
 
-	if !sysTransFlag && tx.GasPrice != 0 {
+	isCharge := !sysTransFlag && tx.GasPrice != 0
+	fmt.Println("isCharge:", isCharge)
```
Comment (Arbio5zt) [inline core/store/ledgerstore/tx_handler.go]: need to change fmt print to debug print.

File: core/store/ledgerstore/tx_handler.go [hunk 4]
```diff
@@ -119,19 +112,18 @@ func (self *StateStore) HandleInvokeTransaction(store store.LedgerStore, stateBa
...
 		CloneCache: cache,
 		Store:      store,
-		Code:       invoke.Code,
 		Gas:        tx.GasLimit,
 	}
 
 	//start the smart contract executive function
-	_, err := sc.Execute()
-
+	engine, _ := sc.NewExecuteEngine(invoke.Code)
```
Comment (Arbio5zt) [inline core/store/ledgerstore/tx_handler.go]: miss error catch.

File: core/store/ledgerstore/tx_handler.go [hunk 5]
```diff
@@ -210,19 +180,17 @@ func isBalanceSufficient(tx *types.Transaction, stateBatch *statestore.StateBatc
...
 		Config:     config,
 		CloneCache: cache,
 		Store:      store,
-		Code:       nativeTransferCode,
 		Gas:        math.MaxUint64,
 	}
 
-	_, err := sc.Execute()
-
+	service, _ := sc.NewNativeService()
```
Comment (Arbio5zt) [inline core/store/ledgerstore/tx_handler.go:195]: miss error catch.

File: vm/neovm/func_array.go (hunk 1 above)
Comment (tanZiWen) [reply vm/neovm/func_array.go]: fix

**Review by Arbio5zt — CHANGES_REQUESTED** [review]

File: vm/neovm/types/array.go [hunk 6]
```diff
@@ -54,31 +54,19 @@ func (this *Array) Equals(other StackItems) bool {
...
-	return this._array[0].GetBigInteger()
+	return big.NewInt(0)
 }
 
 func (this *Array) GetBoolean() bool {
-	if len(this._array) == 0 {
-		return false
-	}
-	return this._array[0].GetBoolean()
+	return true
```
Comment (luodanwg) [inline vm/neovm/types/array.go]: the GetBoolean name means get the bool value, but can only return true ?

**Review by Arbio5zt — APPROVED** [review]

**Pull request merged**